Cython/Compiler/Optimize.py

```diff
@@ -687,7 +747,10 @@ def _transform_range_iteration(self, node, range_function, reversed=False):
         if not bound2.is_literal:
             # stop bound must be immutable => keep it in a temp var
             bound2_is_temp = True
-            bound2 = UtilNodes.LetRefNode(bound2)
+            if bound2_ref_node:
+                bound2 = bound2_ref_node
+            else:
+                bound2 = UtilNodes.LetRefNode(bound2)
```

Comment:
JavaScript's style would have this if-else replaced by `bound2 = bound2_ref_node or UtilNodes.LetRefNode(bound2)`. What's your take on this?